rocky/assets/js/checkboxToggler.js

```diff
@@ -1,4 +1,31 @@
 const toggle_all_btn = document.querySelectorAll(".toggle-all");
+
+function SetToggleONWhenAllChecked() {
+  var count_checked = 0;
+  checkbox_target = document.querySelector(".toggle-all[data-toggle-target]");
+  if (checkbox_target) {
+    var checkboxes = document.getElementsByName(
+      checkbox_target.dataset.toggleTarget,
+    );
+
+    for (var i = 0; i < checkboxes.length; i++) {
+      if (
+        checkboxes[i].tagName == "INPUT" &&
+        checkboxes[i].type == "checkbox"
+      ) {
+        if (checkboxes[i].checked) {
+          count_checked++;
+        }
+      }
+    }
+
+    if (count_checked == checkboxes.length) {
+      checkbox_target.classList.toggle("toggle-on");
+    }
```

Comment:
The `toggle()` method will also remove the class if it's already present. Is that desired behaviour?

If not, you could consider using [`toggle()`'s `force` param](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle#force), either to make sure it's only ever added:

```suggestion
    if (count_checked == checkboxes.length) {
      checkbox_target.classList.toggle("toggle-on", true);
    }
```

...or by using `count_checked == checkboxes.length` directly:

```suggestion
    checkbox_target.classList.toggle("toggle-on", count_checked == checkboxes.length);
```

Reply:
Oeehhh nice parameters I can use indeed to get the desired results. thanks!